doc/guides/west/manifest.rst

```diff
@@ -1179,6 +1189,433 @@ processed in this order:
+<west-manifest-files>` to place projects into groups. The ``groups`` key can be
+a project or top level manifest attribute, like this:
+
+.. code-block:: yaml
+
+   manifest:
+     projects:
+       - name: some-project
+         groups: ...
+     groups: ...
+
+You can allow or block project groups. Projects whose groups are all blocked
+are *inactive*; west essentially ignores inactive projects unless explicitly
+requested not to.
+
+Project Groups
+==============
+
+Inside ``manifest: projects:``, you can add a project to one or more groups.
+The ``groups`` key in this case is a list of group names. Group names are
+strings.
+
+For example, in this manifest fragment:
+
+.. code-block:: yaml
+
+  manifest:
+    projects:
+      - name: project-1
+        groups:
+          - groupA
+      - name: project-2
+        groups:
+          - groupB
+          - groupC
+      - name: project-3
+
+The projects are in these groups:
+
+- ``project-1``: one group, named ``groupA``
+- ``project-2``: two groups, named ``groupB`` and ``groupC``
+- ``project-3``: no groups
+
+Project group names must not contain commas (,), colons (:), or whitespace.
+
+Group names must not begin with a dash (-), but they may contain dashes
+elsewhere in their names: ``foo-bar`` is a valid group name, but ``-foo-bar``
+is not.
+
+Group names are otherwise arbitrary strings. Group names are case sensitive.
+
+As a restriction, no project may use both ``import:`` and ``groups:``. (This
+avoids some edge cases whose semantics are difficult to specify.)
+
+Allowed and Blocked Project Groups
+==================================
+
+You can allow or block project groups in both your manifest file and
+:ref:`west-config`.
+
+All groups are allowed by default.
+
+Within the manifest file, the top level ``manifest: groups:`` value can be used
+to block project groups. To block a group, prefix its name with a dash (-).
+For example, in this manifest fragment:
+
+.. code-block:: yaml
+
+   manifest:
+     groups: [-groupA,-groupB]
+
+The groups named ``groupA`` and ``groupB`` are blocked.
+
+This ``groups`` list is an ordinary YAML list, so you could have also written
+this fragment like this:
+
+.. code-block:: yaml
+
+   manifest:
+     groups:
+       - -groupA
+       - -groupB
+
+However, this syntax is harder to read and therefore discouraged.
+
+Only the **top level manifest file's** ``manifest: groups:`` value has any
+effect. The ``manifest: groups:`` values in any :ref:`imported manifests
+<west-manifest-import>` are ignored.
+
+In addition to the manifest file, you can control the project group allowlist
+and blocklist using the ``manifest.groups`` configuration option. This option
+is a comma-separated list of groups. To allow a group, add its name to the
+list. To block a group, add its name prefixed by a dash (``-``). For example,
+setting ``manifest.groups`` to ``groupA,-groupB`` allows ``groupA``, and blocks
+``groupB``.
```

Comment:
Unless it's too late but I think `+groupA,-groupB` would be more "consistent" than `groupA,-groupB` and it would also make a clearer difference from the `groups` homonym at the project level. It may even be easier to script config files changes if anyone is into that.

Reply:
> Unless it's too late but I think `+groupA,-groupB` would be more "consistent" than `groupA,-groupB` and it would also make a clearer difference from the `groups` homonym at the project level. It may even be easier to script config files changes if anyone is into that.

I would like @carlescufi and @tejlmand to weigh in here. It's not too late, but we should make a decision quickly if we can.

Reply:
> I would like @carlescufi and @tejlmand to weigh in here. It's not too late, but we should make a decision quickly if we can.

I like the proposal from @marc-hb.
Explicit having to do `+groupA` also remove the stupid user mistake of forgetting `-`, so he accidentially does `groupB` when he meant `-groupB`, `groupB` would throw an error at the user.
This greatly reduces one of those *doh* mistakes that can take time to track down.

Reply:
This alternate proposal requires us to forbid group names from starting with `+` or `-` -- are we OK with that?

Reply:
Sorry, to be more clear: they already can't start with `-`, but this would add `+` to the list of forbidden characters. I personally don't have issues with that, but want to get it out in front of people.

Reply:
Yes, I also agree with this proposal, it is more consistent now that we've settled on using this add/remove list of projects. Let's go with it.

Reply:
still agree to use `+groupA` and `-groupB` and thus restrict the use of `+` as first char of a group name.